src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/OldAndBrokenSingletonLocatorAnalyzer.cs

```diff
@@ -47,28 +47,40 @@ INamedTypeSymbol singletonLocatorType
 				return;
 			}
 			var symbolinfo = context.SemanticModel.GetSymbolInfo( root );
+
 			var method = symbolinfo.Symbol as IMethodSymbol;
+
 			if( method == null ) {
-				if( symbolinfo.CandidateSymbols != null && symbolinfo.CandidateSymbols.Length == 1 ) {
-					//This happens on method groups, such as
-					//  Func<IFoo> fooFunc = OldAndBrokenServiceLocator.Get<IFoo>;
-					method = symbolinfo.CandidateSymbols.First() as IMethodSymbol;
-				} else {
+				if( symbolinfo.CandidateSymbols == null ) {
+					return;
+				}
+
+				if( symbolinfo.CandidateSymbols.Length != 1 ) {
+					return;
+				}
+
+				//This happens on method groups, such as
+				//  Func<IFoo> fooFunc = OldAndBrokenServiceLocator.Get<IFoo>;
+				method = symbolinfo.CandidateSymbols.First() as IMethodSymbol;
+
+				if( method == null ) {

```

Comment:
This seems to happen sometimes during keyboard typing. Didn't dig in too much as to why.